torch/_functorch/_aot_autograd/runtime_wrappers.py

```diff
@@ -1712,6 +1711,13 @@ def process_runtime_tangent(x, meta: Union[PlainTensorMeta, SubclassCreationMeta
             expected_meta = meta.meta
 
         runtime_type = type(x)
+        if torch._dynamo.compiled_autograd.in_compiled_autograd_region:
+            # When we're inside compiled autograd's AOTDispatcher step,
+            # regular Tensors look like FunctionalTensors.
+            # Tensor subclasses still look like Tensor subclasses though.
+            if isinstance(x, torch._subclasses.functional_tensor.FunctionalTensor):
+                runtime_type = torch.Tensor
```

Comment:
just calling out, but looks like `process_runtime_tangent()` gets called recursively on the inner tensors in the case where `x` is a subclass, so this should cover the case where you have e.g. `Subclass(Subclass(FunctionalTensor(FakeTensor)))`